pkg/credentials/iam_aws.go

```diff
@@ -82,7 +83,11 @@ func (m *IAM) Retrieve() (Value, error) {
 	case len(os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")) > 0:
 		if len(endpoint) == 0 {
 			if len(os.Getenv("AWS_REGION")) > 0 {
-				endpoint = "https://sts." + os.Getenv("AWS_REGION") + ".amazonaws.com"
+				if strings.HasPrefix(os.Getenv("AWS_REGION"), "cn") {
```

Comment:
`cn-` instead of `cn`

Reply:
By the way, normally you can pass endpoint using NewIAM(endpoint string) API, that would be enough

Reply:
Thanos is one of the client neither set endpoint while creeating IAM and nor using NewIAM function.
May be Thanos should consider to fix this issue also from Thanos side.

Reply:
> `cn-` instead of `cn`

Have considered this review point and fixed.